sdk/communication/communication-chat/src/models/mappers.ts

```diff
@@ -48,14 +60,17 @@ export const mapToChatContentSdkModel = (
 };
 
 /**
+ * @internal
  * Mapping chat message REST model to chat message SDK model
  */
 export const mapToChatMessageSdkModel = (chatMessage: RestModel.ChatMessage): ChatMessage => {
-  const { content, senderId, ...otherChatMessage } = chatMessage;
+  const { content, senderCommunicationIdentifier, ...otherChatMessage } = chatMessage;
   const contentSdkModel = content ? mapToChatContentSdkModel(content) : undefined;
-  if (senderId) {
+  if (senderCommunicationIdentifier) {
     return {
-      sender: { communicationUserId: senderId! },
+      sender: deserializeCommunicationIdentifier(
```

Comment:
`deserializeCommunicationIdentifier` already returns the `CommunicationIdentifierKind`. So it seems like you only need to change `CommunicationIdentifier` to `CommunicationIdentifierKind` in the interfaces that you return from methods or that are part of events, and this will already work.